File: src/visual-graph/builder/is-branch-scope.ts
import type { SerializedScope } from "../../ir/model.js";
import { branchContainerKey } from "./branch-container-key.js";

export function isBranchScope(
  scopeId: string,
  scopeMap: ReadonlyMap<string, SerializedScope>,
): boolean {
  const scope = scopeMap.get(scopeId);
  return scope ? branchContainerKey(scope) !== null : false;
}

File: crates/unsnarl-visual-graph/src/builder/is_branch_scope.rs
use std::collections::HashMap;

use unsnarl_ir::serialized::SerializedScope;

use super::branch_container_key::branch_container_key;

pub fn is_branch_scope(scope_id: &str, scope_map: &HashMap<&str, &SerializedScope>) -> bool {
    scope_map
        .get(scope_id)
        .copied()
        .is_some_and(|s| branch_container_key(s).is_some())
}

#[cfg(test)]
#[path = "is_branch_scope_test.rs"]
mod is_branch_scope_test;
